### ISSUE TITLE
customize whitelist......

### ISSUE THREAD

**Issue by 4evermaat** [Hau-Hau/firefox-tab-suspender#9]
I need to be able to quickly whitelist a tab so that it does not get suspended.   Why is this not possible?  You cannot assume each tab that is opened needs to be suspended.   There must be a way for the user to make a custom determination of which urls (and/or domains) to avoid suspending.

There are a lot of features missing from `The Great Suspender`.

**Comment by pjz** [comment]
Agreed, it would be nice if the extension could add an "Whitelist this from suspension" menu option to the right-click menu on tabs or on the page.

**Comment by Hau-Hau** [comment]
Hi guys!

Whitelist feature is on the way, but because of that the most part of time I develop whole extension by myself, every new feature (even simplest) or bug-fix is very time consuming.

For now I working on non-native way of discarding tabs because some people told me about some bugs with native discarding.

Whitlist feature is next in queue,  but first I need to fix bugs and problems that interfere in  using of extension.

Development will be faster only with other collaborators, for me it isn't possible to work faster - this is 'after work' project.

And yes, i know that there is many missing features, but one after another i will be add next features.

Thanks for your messages and following of my work.

Best regards,
Michalewicz Piotr

**Comment by 4evermaat** [comment]
so did you close this because it is completed?  where is the reference to the commit?

**Comment by Hau-Hau** [comment]
Hi @4evermaat,

I closed this issue because whitelist is feature that was be always in roadmap ([roadmap](https://trello.com/b/BbbjVfl4/firefox-tab-suspender-public)).

But you are right, i will reopen this issue to moment when i finish this feature.

Best regards,
Michalewicz Piotr

**Comment by niladam** [comment]
Acting like jerks won't speed up this process. This is an open source program that's being handled on the dev's own spare time. If you feel like this isn't entirely complete for you, grab your editor and start helping out.

For future reference, a feature request should be exactly that. A **request** and not a comparison or a lame critique.

@Hau-Hau - do your best buddy -- whenever you have the chance. Work when you can, how you can. I'm grateful for your work and really, want to thank you for the time spent on building it **and open sourcing** it!:)

Keep it up.

**Comment by 4evermaat** [comment]
@niladam fair enough.   I thought the author got his influence from "The Great Suspender" chrome extension, and was looking to duplicate its most prominent features on here.

**Comment by niladam** [comment]
@4evermaat he clearly says that's the inspiration. However, inspiration doesn't mean an exact copy. He's working on building those as well -- i guess they're useful to him as well:)

I'm happy to hear you didn't mean to insult, but that's how it sounded :)

**Comment by Hau-Hau** [comment]
Hello!

@niladam 
Thanks for that what you said!
I'm appreciated that my work is useful for other persons!

@4evermaat 
There isn't even one line of code which is copied from any other project. 
Everything in project I'm making from scratch, _The Great Suspender_ was only inspiration to making similar piece of software for Firefox.
_The Great Suspender_ and _Firefox Tab Suspender_ shares only concept, even core technical stuffs are completely different. 
So this is also one of reason of slow development process.

Thanks guys for support! :)

Peace!
Michalewicz Piotr
